eureka-client/src/test/java/com/netflix/discovery/DiscoveryClientRegisterUpdateTest.java

```diff
@@ -0,0 +1,70 @@
+package com.netflix.discovery;
+
+import com.netflix.appinfo.ApplicationInfoManager;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.MyDataCenterInstanceConfig;
+import com.netflix.config.ConfigurationManager;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.UUID;
+
+/**
+ * @author David Liu
+ */
+public class DiscoveryClientRegisterUpdateTest {
+
+    private ApplicationInfoManager applicationInfoManager;
+    private MockRemoteEurekaServer mockLocalEurekaServer;
+
+    @Before
+    public void setUp() throws Exception {
+        mockLocalEurekaServer = new MockRemoteEurekaServer();
+        mockLocalEurekaServer.start();
+
+        ConfigurationManager.getConfigInstance().setProperty("eureka.name", "EurekaTestApp-" + UUID.randomUUID());
+        ConfigurationManager.getConfigInstance().setProperty("eureka.registration.enabled", "true");
+        ConfigurationManager.getConfigInstance().setProperty("eureka.appinfo.replicate.interval", 10);
+        ConfigurationManager.getConfigInstance().setProperty("eureka.shouldFetchRegistry", "false");
+        ConfigurationManager.getConfigInstance().setProperty("eureka.serviceUrl.default",
+                "http://localhost:" + mockLocalEurekaServer.getPort() +
+                        MockRemoteEurekaServer.EUREKA_API_BASE_PATH);
+
+        applicationInfoManager = ApplicationInfoManager.getInstance();
+        applicationInfoManager.initComponent(new MyDataCenterInstanceConfig());
+        EurekaClient client = new DiscoveryClient(applicationInfoManager.getInfo(), new DefaultEurekaClientConfig());
+    }
+
+    @Test
+    public void registerUpdateLifecycleTest() throws Exception {
+        applicationInfoManager.setInstanceStatus(InstanceInfo.InstanceStatus.UP);
+        Thread.sleep(1000);  // give some execution time
```

Comment:
Can these all sleeps be replaced with active pooling? This would speed up the test a lot. I have applied this strategy in replication tests, and they execute pretty fast.